cmd/util/ledger/migrations/change_contract_code_migration.go

```diff
@@ -0,0 +1,342 @@
+				systemContracts.FungibleToken.Address.HexWithPrefix(),
+				systemContracts.FlowToken.Address.HexWithPrefix(),
+				systemContracts.IDTableStaking.Address.HexWithPrefix(),
+				stakingProxyAddress.HexWithPrefix(),
+				lockedTokensAddress.HexWithPrefix(),
+				systemContracts.FlowStorageFees.Address.HexWithPrefix(),
+				systemContracts.ClusterQC.Address.HexWithPrefix(),
+				systemContracts.DKG.Address.HexWithPrefix(),
+				systemContracts.Epoch.Address.HexWithPrefix(),
+			)),
+		},
+		{
+			Address:         stakingProxyAddress,
+			ContractName:    "StakingProxy",
+			NewContractCode: string(coreContracts.FlowStakingProxy()),
+		},
+		{
+			Address:      lockedTokensAddress,
+			ContractName: "LockedTokens",
+			NewContractCode: string(coreContracts.FlowLockedTokens(
+				systemContracts.FungibleToken.Address.HexWithPrefix(),
+				systemContracts.FlowToken.Address.HexWithPrefix(),
+				systemContracts.IDTableStaking.Address.HexWithPrefix(),
+				stakingProxyAddress.HexWithPrefix(),
+				systemContracts.FlowStorageFees.Address.HexWithPrefix(),
+			)),
+		},
+
+		// token related contracts
+		NewSystemContractChange(
+			systemContracts.FlowFees,
+			coreContracts.FlowFees(
+				systemContracts.FungibleToken.Address.HexWithPrefix(),
+				systemContracts.FlowToken.Address.HexWithPrefix(),
+				systemContracts.FlowStorageFees.Address.HexWithPrefix(),
+			),
+		),
+		NewSystemContractChange(
+			systemContracts.FlowToken,
+			coreContracts.FlowToken(
+				systemContracts.FungibleToken.Address.HexWithPrefix(),
+				systemContracts.MetadataViews.Address.HexWithPrefix(),
+				systemContracts.ViewResolver.Address.HexWithPrefix(),
+			),
+		),
+		NewSystemContractChange(
+			systemContracts.FungibleToken,
+			ftContracts.FungibleToken(),
+		),
+		{
+			Address:      fungibleTokenMetadataViewsAddress,
+			ContractName: "FungibleTokenMetadataViews",
+			NewContractCode: string(ftContracts.FungibleTokenMetadataViews(
+				systemContracts.FungibleToken.Address.HexWithPrefix(),
+				systemContracts.MetadataViews.Address.HexWithPrefix(),
+			)),
+		},
+		{
+			Address:      fungibleTokenSwitchboardAddress,
+			ContractName: "FungibleTokenSwitchboard",
+			NewContractCode: string(ftContracts.FungibleTokenSwitchboard(
+				systemContracts.FungibleToken.Address.HexWithPrefix(),
+			)),
+		},
+
+		// NFT related contracts
+		NewSystemContractChange(
+			systemContracts.NonFungibleToken,
+			nftContracts.NonFungibleToken(),
+		),
+		NewSystemContractChange(
+			systemContracts.MetadataViews,
+			nftContracts.MetadataViews(
+				sdk.Address(systemContracts.FungibleToken.Address),
+				sdk.Address(systemContracts.NonFungibleToken.Address),
+			),
+		),
+		NewSystemContractChange(
+			systemContracts.ViewResolver,
+			nftContracts.Resolver(),
+		),
+
+		// EVM related contracts
+		NewSystemContractChange(
+			systemContracts.EVM,
+			evm.ContractCode(
+				systemContracts.FlowToken.Address,
+			),
+		),
```

Comment:
this one is not deployed yet. We should remove it from this list.

Reply:
Could we gate it behind the feature flag that enables EVM?

Reply:
unfortunately the flag is provided as the node startup parameters, so that would means we have to also provide them as migration parameters. But we know the state at the time of the migration, so I think it makes sense to just hard-code this.

Reply:
I think in any case, the version of flow-go with Cadence 1.0 will have the EVM contract, right? Even if it is "deployed" and the EVM functionality is disabled, that's OK, it will just not be usable even though it exists

Reply:
2 days ago I would have said no, but it looks like the EVM contract will actually get deployed very soon (before the next HCU) so my original comment is actually not valid anymore :)